### ISSUE TITLE
Hydrogen stops working with 1.19.0

### ISSUE THREAD

**Issue by DavidLGoldberg** [nteract/hydrogen#938]
**Description:**

After recent upgrade to 1.19.0 Hydrogen commands do not work and disappear from commands (in cmd+shift+p)

Console shows:
![image](https://user-images.githubusercontent.com/578479/29178066-768c6202-7dbe-11e7-81ec-3551a73f775f.png)

After `apm rebuild` inside ~/.atom/packages/Hydrogen still seeing same results.

Is anyone's Hydrogen working with 1.19.0?

**Steps to Reproduce:**

1. Upgrade to Atom 1.19.0

**Versions:**

Atom    : 1.19.0
Electron: 1.6.9
Chrome  : 56.0.2924.87
Node    : 7.4.0

Mac OS X 10.11.6

**Logs:**

Failed to require the main module of 'Hydrogen' because it requires one or more incompatible native modules (zeromq).
Run `apm rebuild` in the package directory and restart Atom to resolve.

**Comment by DavidLGoldberg** [comment]
Did that too, although I think it has the same effect as jumping into the package.  Still same scenario.  I've never seen a command get deregistered that way in Atom EVER.  Really, really strange.  It's probably not *really* a hydrogen issue at all from my experience with this kind of thing. :\  Maybe I can link to this on the atom discussion boards tonight as I've seen a lot of other upgrade -> package rebuild issues recently.

**Comment by DavidLGoldberg** [comment]
BTW, I've been using this at/for work stuff. I was actually right about to give a demo when it stopped working :\ hah!

Anyway, I of course still *really* love the package, might try to contrib later on if I have any ideas.

There's no hydrogen -> notebook feature yet right?

I've been using explicit # %% cells as well as highlight and run (awesome feature!).

**Comment by DavidLGoldberg** [comment]
Cut and pasting into a notebook before a demo or something will totally suffice for now......

**Comment by lgeiger** [comment]
You may want to try completely reinstalling Hydrogen:
- Fully close Atom
- Run `apm uninstall hydrogen`
- Run `apm install hydrogen`

After that hopefully everything is back to normal. Atom can sometimes get in a weird state when it fails to rebuild a package.

**Comment by lgeiger** [comment]
Thanks for the great feedback!

> Anyway, I of course still really love the package, might try to contrib later on if I have any ideas.

🎉 That's awesome! You may want to take a look at our [`new-contributor-friendly` issues](https://github.com/nteract/hydrogen/issues?q=is%3Aissue+is%3Aopen+label%3Anew-contributor-friendly) and feel free to join our [Slack](https://slackin-nteract.now.sh/) if you have any questions.

> There's no hydrogen -> notebook feature yet right?

We have a open issue with more information but no one is currently working it: #801

**Comment by DavidLGoldberg** [comment]
Unfortunately closing / uninstalling didn't fix it.  Although I'm embarrassed to say I forgot trying that!

I'll try a compete reinstall of Atom tonight.  Might be overdue / why some of these things are occurring.

Just want to triple confirm you're on 1.19.0 right with a restart and things are working on your end?

Pretty sure you have enough users you would have a bunch of people weighing in.  Probably just my install is corrupt :\

I'll let you know tonight how this went.

**Comment by BenRussert** [comment]
> Just want to triple confirm you're on 1.19.0 right with a restart and things are working on your end?

I can confirm this from my end (though I am on ubuntu).

I dont know why the `apm rebuild hydrogen` solution would not have the same result, but I just want to point out a common problem and solution. See also #937 to see if your problem is similar.

Each time you install a new version of atom there will be an [incompatible packages](https://github.com/atom/incompatible-packages/blob/master/README.md) view with an overview of incompatible native module dependencies. These issues can be typically be resolved by clicking a button that says "rebuild modules".

Hope that helps!

**Comment by jlowin** [comment]
For what it's worth, I have the same issue after updating Atom. I get the same error: `Failed to require the main module of 'Hydrogen' because it requires one or more incompatible native modules (zeromq). Run apm rebuild in the package directory and restart Atom to resolve.`

I've reinstalled Hydrogen, rebuilt Hydrogen, even reinstalled Atom -- still see the error. 😢

**Comment by BenRussert** [comment]
Awesome! Glad it worked.

I was curious why the gui method works, but not the cli. Based on the [incompatible packages source](https://github.com/atom/atom/blob/master/src/package.coffee#L738-L748), this *should* be the same as cd'ing to the package directory where hydrogen is installed (most likely `~/.atom/packages/Hydrogen/` or platform equivalent, then running `apm rebuild` (more precisely `apm rebuild --no-color`)

A possible exception is for Mac users on atom beta, in which case you need to run `apm-beta rebuild` from the dir where Hydrogen is installed.

**Comment by DavidLGoldberg** [comment]
Sorry for the slow response, fyi:

I had forgotten what the GUI version was called...I kept trying "rebuild" in cmd+shift+p instead of "incompatible" :-\  Yes that is and has been totally necessary (instead of the command line!)

When I ran it, I didn't get Hydrogen listed though.  I had 2 other unrelated packages I had to fix.

Unfortunately I didn't get to test Hydrogen immediately after, before upgrading to 1.19.1.

So either the incompatible packages GUI (probably) or the upgrade fixed it.

Either way, All working again!  Thanks everyone!
